tools/ci_build/github/azure-pipelines/mac-ios-ci-pipeline.yml

```diff
@@ -58,3 +58,33 @@ jobs:
           --path_to_protoc_exe $(Build.BinariesDirectory)/protobuf_install/bin/protoc \
           --parallel
       displayName: (CoreML EP) Build onnxruntime for iOS x86_64 and run tests using simulator
+
+- job: iOS_CI_XNNPACK_on_Mac
```

Comment:
Do we really need to run this job for every pull request? We have limited macOS machines available, and this build pipeline already takes long.

Reply:
What if this job will be running when we trigged it or merged it into main branch

Reply:
That would be better.

Reply:
If ORT has 10 cmake options, then there would be 2^10=1024 different combinations. Our build pipelines can't cover all of them, and we have far more than 10 build options(or probably more than 100). So I suggest focusing on the things we ship. For example, if every iOS package we release contains this XNNPack EP, then we may consider to adjust our pull request pipelines to match it. Meanwhile, you may have extra pipelines to test the combinations that you think are important, but because of resource limitations we can't trigger all such pipelines for all pull requests.

Reply:
In this way, maybe we can combine multi build pipelines into one for pull-request. Take IOS-build as example, we can only build once for (CPU-ep,CORE-ep,XNNPACK-ep), How do you think?